astropy/io/fits/tests/test_image.py

```diff
@@ -100,6 +100,30 @@ def test_open_2(self):
         finally:
             r.close()
 
+    def test_open_3(self):
+        # Test that HDUs cannot be accessed after the file was closed
+        r = fits.open(self.data('test0.fits'))
```

Comment:
Should we also test the context manager (since that is how we recommend open be done anyway, or at least I do) or does it not matter?